File: 백준/Silver/2941. 크로아티아 알파벳/크로아티아 알파벳.c
#include<stdio.h>
int main()
{
	char a[101];
	int cnt=1, i=0;
	while(1){
		scanf("%c", &a[i]);
		if (a[i] == '\n')
			break;
		i++;
	}
	for (int j = 1; j < i; j++)	{
		if (a[j] == '=')
		{
			if (a[j - 1] == 'z' && a[j - 2] == 'd')
				cnt -= 2;
			else if (a[j-1]=='c'|| a[j - 1] == 's'|| a[j - 1] == 'z')
				cnt--;
		}
		else if (a[j] == '-')
		{
			if (a[j - 1] == 'c' || a[j - 1] == 'd')
				cnt--;
		}
		else if (a[j] == 'j')
		{
			if (a[j - 1] == 'l' || a[j - 1] == 'n')
				cnt--;
		}
		cnt++;
	}
	printf("%d", cnt);
}

File: 백준/Silver/2941. 크로아티아 알파벳/크로아티아 알파벳.java
import java.util.Scanner;

public class Main {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		String s = sc.nextLine();
		
		int cnt=0;
		for(int i=0; i<s.length(); i++) {
			cnt++;
			if(i==s.length()-1) //런타임에러 주의. 마지막 글자면 아래에 해당되지 않으므로 빠져나감
				break;
			if (s.charAt(i)=='c') {
				if(s.charAt(i+1)=='='||s.charAt(i+1)=='-') {
					i++;
				}
			}
			
			else if (s.charAt(i)=='d') {
				if (s.charAt(i+1)=='z') {
					if (i<s.length()-2&&s.charAt(i+2)=='=') { // 런타임에러 주의
						i+=2;
					}
				}
				else if (s.charAt(i+1)=='-') {
					i++;
				}
			}
			
			else if (s.charAt(i)=='l') {
				if (s.charAt(i+1)=='j') {
					i++;
				}
			}
			
			else if (s.charAt(i)=='n') {
				if (s.charAt(i+1)=='j') {
					i++;
				}
			}
			
			else if (s.charAt(i)=='s') {
				if (s.charAt(i+1)=='=') {
					i++;
				}
			}
			
			else if (s.charAt(i)=='z') {
				if (s.charAt(i+1)=='=') {
					i++;
				}
			}
		}
		System.out.println(cnt);
		sc.close();
	}

}
